### DISCUSSION TITLE
Product ideas

### DISCUSSION THREAD

**Discussion by joemasilotti** [joemasilotti/railsdevs.com#174] [Ideas]
I'm going to use this discussion to keep track of new product ideas that aren't fleshed out enough to become their own discussion. Ideally, ideas will "graduate" from here to their own project or discussion when we decide to work on them.

**Comment by joemasilotti** [comment]
## Filtering

We started talking about this in #61 and I'd love to keep kicking that can. Here are three specific filters requested from a business:

1. Available now (and also, "soon")
2. Rate
3. Available hours
4. Time zone
5. Location (different than time zone?)

The last one would be a new field on the developer profile, too.

**Reply by joemasilotti** [reply]
I started a spike on this in PR #177 using a `Query` PORO.

**Reply by salochara** [reply]
Hey 👋 
I'm wondering if remote/in-person job filter isn't relevant for business owners? 🤔 

At least from a developer perspective, I do believe that's relevant

**Reply by joemasilotti** [reply]
Yeah, that's a good point. If you don't line up with who's hiring it's kind of a non-starter. Thanks for the idea!

**Comment by joemasilotti** [comment]
## Contracts and payment

In an ideal world a business could hire a developer without ever leaving the site. That includes sending them a contract, both parties signing the contract, and handling payment and payout.

**Comment by joemasilotti** [comment]
## Importance of certain fields

Only a few developers have filled out the expected compensation section on their profile. I'd love to send an email to everyone who signed up explaining that businesses are filtering by these fields, so if you want to get hired make sure you fill them out.

**Comment by joemasilotti** [comment]
## Information showing/hiding for subscribers

Ideally, social links should only be shown to folks with business subscriptions. This is to combat people getting around paying by reaching out to developers via other means.

Also, showing the developer's first name both in their profile and on the new conversation screen will go a long way in making first contact more seamless.

**Reply by joemasilotti** [reply]
Issue created and prioritized with #189 and #188.

**Reply by joemasilotti** [reply]
First pass done, thanks to @jacobdaddario!

Now that paywalled content is hidden, should we tease it to encourage businesses to sign up? Here's a rough pass at what that could look like. I'm a little worried it's too aggressive, but I also love the idea of making the business subscription more obviously valuable.

<img width="1072" alt="image" src="https://user-images.githubusercontent.com/2092156/147898172-57e68040-b4af-41dc-9dea-052e4a76130c.png">

Also, I'm kind of in love with this little affordance!

https://user-images.githubusercontent.com/2092156/147898296-558f97b5-82a7-4b8a-a04d-28f5474b0cfd.mov

More details in PR #206, if you're curious.

**Comment by joemasilotti** [comment]
## Billing

Link to the Stripe Account Portal from the user drop down for users with an associated Pay customer.

**Reply by joemasilotti** [reply]
Issue created and prioritized with #190.

**Reply by joemasilotti** [reply]
Done!

**Comment by freebian** [comment]
### Enhance profiles with structured data
Looking at the profiles I noticed that many developers add data attributes to their profile description, such as:
- Frameworks
- Database
- Cloud
- etc.

I think it would make sense to make this more structured so companies could also filter for it?

Moreover - I know you were asking this on Twitter - many developers add their working experience in years to the profile. This could also be a "smart" feature: Instead of putting in a number of years you put in the year you started. Hence the years are automatically updated.

Additionally, data points that could make sense?
- Industry: Maybe interesting for companies to know if someone built an application in banking or health?
- Brands: In my field (UX) it is always good to know if someone did a project for major brand XYZ. Could also be translated into a nice "Logo Wall" in terms of profile layout.

Regarding implementation I see two different approaches:
1. One model for each attribute
2. Working with something like https://github.com/mbleigh/acts-as-taggable-on

In both ways we should consider that there might be attributes we cannot define beforehand such as brands.

What do you think?

**Reply by joemasilotti** [reply]
I think there's something in tags or expertise, but I'm not sure how I want to expose it just yet. I don't think freeform works because how would a business filter by it if there were 300+ tags to scroll through? I think a more structured "top 30" or somewhat list might make more sense. But choosing those will be tough.

I love the idea of showing off your past work and experience. I've been thinking about adding more official "verified" quotes or similar to the site for past projects worked on via railsdevs.
